lib/iris/experimental/ugrid/mesh.py

```diff
@@ -1631,6 +1662,39 @@ def coords(
             :class:`Mesh` that matched the given criteria.
 
         """
+        if location is not None:
```

Comment:
I can understand why philosophically you are performing the validation here, but it would be cleaner to pass straight down to the underlying filtering methods. There is already some handling there:

https://github.com/SciTools/iris/blob/96cb02600b52a18ae3c30b8b145dd927d57a84fd/lib/iris/experimental/ugrid/mesh.py#L2279-L2284

Would that also offer an opportunity to avoid the duplication we currently have?

Reply:
We discussed offline this morning and I still think it's important to push any handling down to this level, especially with the edits to `remove_coords()` as well. The code block I linked to also needs to be removed since the 'tri-state' is no longer something on offer for the user.